util/http/http_client.h

```diff
@@ -128,5 +133,34 @@ template <typename Req> std::error_code Client::Send(const Req& req) {
   return HandleError(ec);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+
+class SecureClient : public Client {
+ public:
+  // see https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_security_level.html
+  enum SecurityLevel {
+    LEVEL_0 = 0,  // == nothing basically
+    LEVEL_1,      // The security level corresponds to a minimum of 80, SSL version 2 is prohibited
+    LEVEL_2,      // Security level set to 112 bits of security, SSL version 3 is prohibited
+    LEVEL_3,      // Security level set to 128 bits of security, TLS version 1.1 and above
+    LEVEL_4,      // Security level set to 192 bits of security, TLS version 1.2 and above
+    LEVEL_5       // Security level set to 256 bits of security
+  };
+
+  // note: try using level 2, this option normally work
+  // out of the box
+  static SSL_CTX* CreateSslContext(SecurityLevel sec_level);
+
+  explicit SecureClient(ProactorBase* proactor) : Client(proactor) {
+  }
+
+  std::error_code Connect(StringPiece host, StringPiece service, SSL_CTX* context);
+
+ private:
+  using SecureSocket = util::tls::TlsSocket;
+
+  std::unique_ptr<FiberSocketBase> connected_socket_;
```

Comment:
maybe call it `tcp_socket_` ?

Reply:
OK, I didn't like the name I come up with :)